### PR TITLE
Suspicious saas access from a tor exit node

### PR DESCRIPTION
**Pull request by ArikDay**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-11357)

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Core/Playbooks/playbook-Suspicious_SaaS_Access_From_a_TOR_Exit_Node.yml [hunk 1]
```diff
@@ -0,0 +1,1185 @@
...
+      script: '|||gsuite-user-signout'
+      type: regular
+      iscommand: true
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "7"
+    scriptarguments:
+      user_key:
+        simple: ${Core.OriginalAlert.event.identity_name}
```
Comment (AdiPeret) [inline Packs/Core/Playbooks/playbook-Suspicious_SaaS_Access_From_a_TOR_Exit_Node.yml:299]: Why not use filters & transformers?

File: Packs/Core/Playbooks/playbook-Suspicious_SaaS_Access_From_a_TOR_Exit_Node.yml [hunk 2]
```diff
@@ -0,0 +1,1185 @@
...
+      script: '|||core-list-risky-users'
+      type: regular
+      iscommand: true
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "9"
+    scriptarguments:
+      user_id:
+        simple: ${alert.username.[0]}
```
Comment (AdiPeret) [inline Packs/Core/Playbooks/playbook-Suspicious_SaaS_Access_From_a_TOR_Exit_Node.yml:366]: see comments above

File: Packs/Core/Playbooks/playbook-Suspicious_SaaS_Access_From_a_TOR_Exit_Node.yml [hunk 3]
```diff
@@ -0,0 +1,1185 @@
...
+      script: '|||msgraph-user-session-revoke'
+      type: regular
+      iscommand: true
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "7"
+    scriptarguments:
+      user:
+        simple: ${MSGraphUser.ID}
```
Comment (AdiPeret) [inline Packs/Core/Playbooks/playbook-Suspicious_SaaS_Access_From_a_TOR_Exit_Node.yml:134]: see  comment above

**Review by AdiPeret — APPROVED** [review]
Looks good

File: Packs/Core/Playbooks/playbook-Suspicious_SaaS_Access_From_a_TOR_Exit_Node.yml (hunk 1 above)
Comment (ArikDay) [reply Packs/Core/Playbooks/playbook-Suspicious_SaaS_Access_From_a_TOR_Exit_Node.yml:299]: Not possible in XSIAM

File: Packs/Core/Playbooks/playbook-Suspicious_SaaS_Access_From_a_TOR_Exit_Node.yml (hunk 2 above)
Comment (ArikDay) [reply Packs/Core/Playbooks/playbook-Suspicious_SaaS_Access_From_a_TOR_Exit_Node.yml:366]: Not possible in XSIAM

File: Packs/Core/Playbooks/playbook-Suspicious_SaaS_Access_From_a_TOR_Exit_Node.yml (hunk 3 above)
Comment (ArikDay) [reply Packs/Core/Playbooks/playbook-Suspicious_SaaS_Access_From_a_TOR_Exit_Node.yml:134]: Not possible in XSIAM

File: Packs/Core/ReleaseNotes/3_0_56.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+Investigation: Checks the user’s risk score. Checks for suspicious user agent usage within the alert.
+Containment: Based on the user’s risk score and user agent, the playbook will suggest to block the account involved.
+Eradication: For users with PAN-OS enabled, the playbook will suggest to block all IPs from Palo Alto Intelligence-based external dynamic list that contains TOR exit nodes.
+Requires: Integrations - Microsoft Graph User / G-Suite Admin.
+
+#### Triggers Recommendations
+
+##### New: Suspicious SaaS Access From a TOR Exit Node
+
+- New: This trigger is responsible for handling Suspicious SaaS Access From a TOR Exit Node<~XSIAM> (Available from Cortex XSIAM).</~XSIAM>
```
Comment (ShirleyDenkberg) [inline Packs/Core/ReleaseNotes/3_0_56.md:23]:
```suggestion
New: This trigger is responsible for handling Suspicious SaaS Access From a TOR Exit Node<~XSIAM> (Available from Cortex XSIAM).</~XSIAM>
```

File: Packs/PAN-OS/Playbooks/playbook-PAN-OS_-_Block_IPs_From_EDL_-_Custom_Block_Rule_README.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
+This playbook blocks IP addresses from External Dynamic List using Custom Block Rules in Palo Alto Networks Panorama or Firewall. The playbook receives EDL name as input, creates a custom "from" directional rule to block, and commits the configuration.
```
Comment (ShirleyDenkberg) [inline Packs/PAN-OS/Playbooks/playbook-PAN-OS_-_Block_IPs_From_EDL_-_Custom_Block_Rule_README.md:1]:
```suggestion
This playbook blocks IP addresses from an External Dynamic List using Custom Block Rules in Palo Alto Networks Panorama or Firewall. The playbook receives an EDL name as input, creates a custom "from" directional rule to block, and commits the configuration.
```

File: Packs/PAN-OS/Playbooks/playbook-PAN-OS_-_Block_IPs_From_EDL_-_Custom_Block_Rule_README.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+This playbook uses the following sub-playbooks, integrations, and scripts.
+
+### Sub-playbooks
+
+This playbook does not use any sub-playbooks.
+
+### Integrations
+
+* Panorama
```
Comment (ShirleyDenkberg) [inline Packs/PAN-OS/Playbooks/playbook-PAN-OS_-_Block_IPs_From_EDL_-_Custom_Block_Rule_README.md:13]:
```suggestion
Panorama
```

File: Packs/PAN-OS/Playbooks/playbook-PAN-OS_-_Block_IPs_From_EDL_-_Custom_Block_Rule_README.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+## Playbook Inputs
+
+---
+
+| **Name** | **Description** | **Default Value** | **Required** |
+| --- | --- | --- | --- |
+| RuleName | Panorama rule name |  | Optional |
+| EDLName | EDL IPs to block |  | Optional |
+| AutoCommit | This input establishes whether to commit the configuration automatically.<br/>Yes - Commit automatically.<br/>No - Commit manually. | No | Optional |
+| LogForwarding | Panorama log forwarding object name |  | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/PAN-OS/Playbooks/playbook-PAN-OS_-_Block_IPs_From_EDL_-_Custom_Block_Rule_README.md:33]:
```suggestion
| RuleName | Panorama rule name. |  | Optional |
| EDLName | EDL IPs to block. |  | Optional |
| AutoCommit | This input establishes whether to commit the configuration automatically.<br/>Yes - Commit automatically.<br/>No - Commit manually. | No | Optional |
| LogForwarding | Panorama log forwarding object name. |  | Optional |
```

File: Packs/PAN-OS/ReleaseNotes/2_1_35.md [hunk 8]
```diff
@@ -0,0 +1,6 @@
+
+#### Playbooks
+
+##### New: PAN-OS - Block IPs From EDL - Custom Block Rule
+
+- New: This playbook blocks IP addresses from External Dynamic List using Custom Block Rules in Palo Alto Networks Panorama or Firewall. The playbook receives EDL name as input, creates a custom "from" directional rule to block, and commits the configuration.
```
Comment (ShirleyDenkberg) [inline Packs/PAN-OS/ReleaseNotes/2_1_35.md:6]:
```suggestion
New: This playbook blocks IP addresses from an External Dynamic List using Custom Block Rules in Palo Alto Networks Panorama or Firewall. The playbook receives an EDL name as input, creates a custom "from" directional rule to block, and commits the configuration.
```

**Comment by ShirleyDenkberg** [comment]
@AdiPeret Doc review completed.

**Pull request merged**